pkg/threescale/backend/manager.go

```diff
@@ -183,9 +187,12 @@ func (b *Backend) Flush() {
 			// for each application for this service
 			// take a read lock and take a snapshot of the current state
 			app.RLock()
+			// take a snapshot of the current application
```

Comment:
can we `RUnlock` right after the `deepCopy` and compute deltas using the snapshot instead of `app`?

Reply:
yes we can. we want to preserve the state of the snapshot but the functionality that follows it does not modify it in this particular case. nice catch

Reply:
fixed